### PR TITLE
Set Yandex as default

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/12979

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Please follow main issue for test plan

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -312,6 +323,16 @@ public void finishNativeInitialization() {
         }
     }
 
+    private void checkForYandexSE() {
+        TemplateUrl yandexTemplateUrl =
+                BraveSearchEngineUtils.getTemplateUrlByShortName(OnboardingPrefManager.YANDEX);
+        String countryCode = Locale.getDefault().getCountry();
+        if (yandexRegions.contains(countryCode) && yandexTemplateUrl != null) {
+            BraveSearchEngineUtils.setDSEPrefs(yandexTemplateUrl, false);
+            BraveSearchEngineUtils.setDSEPrefs(yandexTemplateUrl, true);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:332]: why do we need to set it to false first and after that to true?

File: android/java/org/chromium/chrome/browser/onboarding/SearchEngineOnboardingFragment.java [hunk 2]
```diff
@@ -58,12 +58,13 @@ private void refreshData() {
...
         List<TemplateUrl> templateUrls = templateUrlService.getTemplateUrls();
         TemplateUrl defaultSearchEngineTemplateUrl =
-            templateUrlService.getDefaultSearchEngineTemplateUrl();
+            BraveSearchEngineUtils.getTemplateUrlByShortName(BraveSearchEngineUtils.getDSEShortName(false));
 
         for (TemplateUrl templateUrl : templateUrls) {
             if (templateUrl.getIsPrepopulated()
                     && OnboardingPrefManager.searchEngineMap.get(templateUrl.getShortName())
                     != null) {
+                Log.e("NTP", templateUrl.getShortName());
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/onboarding/SearchEngineOnboardingFragment.java:67]: I don't think we need that log anymore

File: android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java [hunk 3]
```diff
@@ -71,7 +71,7 @@
     private static final String BING = "Bing";
     private static final String STARTPAGE = "Startpage";
     private static final String YAHOO = "Yahoo";
-    private static final String YANDEX = "Yandex";
+    public static final String YANDEX = "Yandex";
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java:74]: not sure why it has been changed from `private` to `public`

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java (hunk 1 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:332]: second parameter for setSDEPrefs is to pass whether tab is  incogtinito or not.

File: android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java (hunk 3 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java:74]: we had to explicitly make yandex as default browser so making the short name as public static.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 2 XL` running `Android 11` using the following build:

```
Brave | 1.20.19 Chromium: 87.0.4280.101 (Official Build) canary (32-bit)
--- | ---
Revision | 9407c80213cda69c2b7abcb4fa8e3f74488f4956-refs/branch-heads/4280@{#1807}
OS | Android 11; Pixel 2 XL Build/RP1A.201005.004
```

* ensured that `Yandex` was selected as the default search engine for the following locales:
  * `Russia`, `Armenia`, `Belarus` & `Kyrgyz Republic`
* ensured that updating from `1.20.17 CR: 87.0.4280.88` --> `1.20.19 Chromium: 87.0.4280.101` didn't replace the previous default SE with `Yandex`

**`Note:`** Some of these regions don't have any localization but I ensured that:
 * `Yandex` is being set as the default
 * `Yandex` is available via the SE onboarding

### **`Russia Examples`**

**`Example 1`**| **`Example 2`**| 
----------------|------------------
Screenshot_20201208-231549 | Screenshot_20201208-231706


### **`Armenia Examples`**

**`Example 1`**| **`Example 2`**| 
----------------|------------------
Screenshot_20201208-232543 | Screenshot_20201208-232534

**Pull request merged**